crates/astria-sequencer-relayer/src/api.rs

```diff
@@ -103,17 +187,19 @@ impl IntoResponse for Readyz {
     }
 }
 
-#[derive(Debug, Serialize)]
-struct Status {
-    current_sequencer_height: Option<u64>,
-    current_data_availability_height: Option<u64>,
-}
-
-impl Status {
-    fn from_relayer_status(relayer_status: &RelayerState) -> Self {
-        Self {
-            current_sequencer_height: relayer_status.current_sequencer_height,
-            current_data_availability_height: relayer_status.current_data_availability_height,
-        }
-    }
+async fn get_number_of_subscribers(tx: UnboundedSender<network::InfoQuery>) -> eyre::Result<u64> {
+    use eyre::Report;
+    use network::InfoQuery;
+    let (info_response_tx, info_response_rx) = oneshot::channel();
+    tx.send(InfoQuery::NumberOfPeers(info_response_tx))
```

Comment:
wow, this seems complex haha, it makes sense though. is this a common method for doing request/responses between tasks?

Reply:
I've seen and done it before when doing computations on a rayon threadpool triggered by requests that came in through the async (tokio) runtime. So sending the response channel to rayon was a way to show that it's computation is complete.

I was about to reach for permanent channels (one for sending requests to gossipnet, one to receive them from gossipnet). But how do you match which response was for which request? So then you have to make an envelope or add a tag or w/e, and store that in a map....

At first glance it feels complex because sending a channel back and forth is unusual, but the solution is much simpler.